lib/synapse/service_watcher/base.rb

```diff
@@ -117,7 +117,7 @@ def set_backends(new_backends)
       # Aggregate and deduplicate all potential backend service instances.
       new_backends = (new_backends + @default_servers) if @keep_default_servers
       new_backends = new_backends.uniq {|b|
-        [b['host'], b['port'], b.fetch('name', '')]
+        [b['host'], b['port'], b.fetch('name', ''), b.fetch('weight', 1)]
```

Comment:
This is slightly surprising behavior to me because it means while folks are switching weights (i.e. old nerve is running with weight=10 and new nerve is running with weight=20) we'd get both backends. I think I prefer not uniqing on weight (so we'll pick one backend or the other). What do you think?